trie/node.go

```diff
@@ -112,13 +113,29 @@ func computeAndSetNodeHash(n node) ([]byte, error) {
 }
 
 func getNodeFromDBAndDecode(n []byte, db common.DBWriteCacher, marshalizer marshal.Marshalizer, hasher hashing.Hasher) (node, error) {
-	encChild, err := db.Get(n)
+	return getNodeFromDBAndDecodeWithEpoch(n, core.OptionalUint32{}, db, marshalizer, hasher)
+}
+
+func getNodeFromDBAndDecodeWithEpoch(n []byte, epoch core.OptionalUint32, db common.DBWriteCacher, marshalizer marshal.Marshalizer, hasher hashing.Hasher) (node, error) {
+	var data []byte
+	var err error
+
+	if epoch.HasValue {
+		dbWithEpoch, ok := db.(dbWithGetFromEpoch)
```

Comment:
This is a temporary workaround, so that this PR contains as little changes as possible.